File: backend/services/sponsor_categories.py
"""Catalogo de categorias de patrocinio.

Es el esquema comercial con el que entra cada patrocinador: cada categoria
tiene sus cupos y su aporte de referencia. Vive en el codigo (no en la base)
porque es el mismo para todas las ediciones; cambiarlo es un commit.

El frontend tiene su copia en `frontend/src/lib/sponsorCategories.js`: los
slugs de las dos listas tienen que coincidir, porque el slug es lo que se
guarda en `sponsors.propuesta_categoria`.

`cupos` en None significa que la categoria no tiene tope (se muestra con
`cupos_label`). `monto` en None es el aporte que no se paga en efectivo.
`es_patrocinio` en False deja fuera de la vitrina de patrocinadores a quien
compra presencia pero no patrocina (Zona de Marcas).
"""

CATEGORIES = [
    {
        "slug": "titulo",
        "label": "Título",
        "subtitle": "Presenting",
        "cupos": 1,
        "cupos_label": "1",
        "monto": 350000,
        "monto_label": "350,000",
        "incluye": "Naming del evento, máxima visibilidad en camiseta, tarima, prensa y transmisión.",
        "es_patrocinio": True,
    },
    {
        "slug": "platino",
        "label": "Platino",
        "subtitle": None,
        "cupos": 3,
        "cupos_label": "3",
        "monto": 175000,
        "monto_label": "175,000",
        "incluye": "Logo en espalda alta, trofeo con nombre de marca (Últ. Hombre / Últ. Mujer), banner y menciones en vivo.",
        "es_patrocinio": True,
    },
    {
        "slug": "oro",
        "label": "Oro",
        "subtitle": None,
        "cupos": 4,
        "cupos_label": "4",
        "monto": 100000,
        "monto_label": "100,000",
        "incluye": "Logo en mangas o en camiseta finisher 7+, logo en cintillo de la medalla, logo en el bulto, banner y redes.",
        "es_patrocinio": True,
    },
    {
        "slug": "plata",
        "label": "Plata",
        "subtitle": None,
        "cupos": 4,
        "cupos_label": "4",
        "monto": 55000,
        "monto_label": "55,000",
        "incluye": "Logo en el cintillo de la medalla, logo en el bulto del corredor, banner y redes.",
        "es_patrocinio": True,
    },
    {
        "slug": "bronce",
        "label": "Bronce",
        "subtitle": None,
        "cupos": 6,
        "cupos_label": "6",
        "monto": 25000,
        "monto_label": "25,000",
        "incluye": "Logo en el bulto del corredor, presencia en web y redes.",
        "es_patrocinio": True,
    },
    {
        "slug": "experiencia_4x4",
        "label": "Experiencia 4x4",
        "subtitle": "Automotriz",
        "cupos": 1,
        "cupos_label": "1",
        "monto": 300000,
        "monto_label": "300,000",
        "incluye": "Categoría exclusiva: activación y pruebas de manejo en pista 4x4 / motocross, banner, redes y menciones.",
        "es_patrocinio": True,
    },
    {
        "slug": "especie",
        "label": "Aliado en Especie",
        "subtitle": None,
        "cupos": None,
        "cupos_label": "s/valor",
        "monto": None,
        "monto_label": "Especie",
        "incluye": "Hidratación, nutrición, servicio médico, cronometraje, tecnología. Se ubica en la categoría equivalente.",
        "es_patrocinio": True,
    },
    {
        "slug": "media_partner",
        "label": "Media Partner",
        "subtitle": None,
        "cupos": None,
        "cupos_label": "libre",
        "monto": None,
        "monto_label": "Especie",
        "incluye": "Cobertura y difusión del evento: prensa, radio, televisión, podcast o medios digitales. Aporte en especie valorado a precio de mercado.",
        "es_patrocinio": True,
    },
    {
        "slug": "zona_marcas",
        "label": "Zona de Marcas",
        "subtitle": None,
        "cupos": None,
        "cupos_label": "libre",
        "monto": 3000,
        "monto_label": "3,000/ev",
        "incluye": "Stand para sampling. Flexible: RD$3,000/evento o RD$12,000 por los 5 eventos. No constituye patrocinio.",
        "es_patrocinio": False,
    },
]

NOTA_AL_PIE = (
    "Precios de referencia sujetos a disponibilidad de cupos. "
    "Los aportes en especie se valoran a precio de mercado."
)

CATEGORIES_BY_SLUG = {c["slug"]: c for c in CATEGORIES}
CATEGORY_SLUGS = tuple(CATEGORIES_BY_SLUG)


def es_valida(slug: str) -> bool:
    """Vacio significa "sin categoria asignada" y tambien es valido."""
    return not slug or slug in CATEGORIES_BY_SLUG


def etiqueta(slug: str) -> str:
    categoria = CATEGORIES_BY_SLUG.get(slug or "")
    return categoria["label"] if categoria else (slug or "")

File: frontend/src/lib/sponsorCategories.js
// Catálogo de categorías de patrocinio: el esquema comercial con el que entra
// cada patrocinador. Es copia de `backend/services/sponsor_categories.py`
// (el backend valida contra esa lista), así que los slugs de las dos tienen
// que coincidir: el slug es lo que se guarda en el patrocinador.
//
// `cupos: null` = sin tope (se muestra `cuposLabel`). `monto: null` = aporte
// que no se paga en efectivo. `esPatrocinio: false` deja fuera de la vitrina
// a quien compra presencia pero no patrocina (Zona de Marcas).

export const SPONSOR_CATEGORIES = [
  {
    slug: 'titulo',
    label: 'Título',
    subtitle: 'Presenting',
    cupos: 1,
    cuposLabel: '1',
    monto: 350000,
    montoLabel: '350,000',
    incluye: 'Naming del evento, máxima visibilidad en camiseta, tarima, prensa y transmisión.',
    esPatrocinio: true,
    badgeClass: 'bg-neutral-900 text-white',
  },
  {
    slug: 'platino',
    label: 'Platino',
    subtitle: null,
    cupos: 3,
    cuposLabel: '3',
    monto: 175000,
    montoLabel: '175,000',
    incluye: 'Logo en espalda alta, trofeo con nombre de marca (Últ. Hombre / Últ. Mujer), banner y menciones en vivo.',
    esPatrocinio: true,
    badgeClass: 'bg-slate-200 text-slate-800',
  },
  {
    slug: 'oro',
    label: 'Oro',
    subtitle: null,
    cupos: 4,
    cuposLabel: '4',
    monto: 100000,
    montoLabel: '100,000',
    incluye: 'Logo en mangas o en camiseta finisher 7+, logo en cintillo de la medalla, logo en el bulto, banner y redes.',
    esPatrocinio: true,
    badgeClass: 'bg-amber-100 text-amber-800',
  },
  {
    slug: 'plata',
    label: 'Plata',
    subtitle: null,
    cupos: 4,
    cuposLabel: '4',
    monto: 55000,
    montoLabel: '55,000',
    incluye: 'Logo en el cintillo de la medalla, logo en el bulto del corredor, banner y redes.',
    esPatrocinio: true,
    badgeClass: 'bg-gray-200 text-gray-700',
  },
  {
    slug: 'bronce',
    label: 'Bronce',
    subtitle: null,
    cupos: 6,
    cuposLabel: '6',
    monto: 25000,
    montoLabel: '25,000',
    incluye: 'Logo en el bulto del corredor, presencia en web y redes.',
    esPatrocinio: true,
    badgeClass: 'bg-orange-100 text-orange-800',
  },
  {
    slug: 'experiencia_4x4',
    label: 'Experiencia 4x4',
    subtitle: 'Automotriz',
    cupos: 1,
    cuposLabel: '1',
    monto: 300000,
    montoLabel: '300,000',
    incluye: 'Categoría exclusiva: activación y pruebas de manejo en pista 4x4 / motocross, banner, redes y menciones.',
    esPatrocinio: true,
    badgeClass: 'bg-emerald-800 text-white',
  },
  {
    slug: 'especie',
    label: 'Aliado',
    subtitle: null,
    cupos: null,
    cuposLabel: 's/valor',
    monto: null,
    montoLabel: 'Especie',
    incluye: 'Hidratación, nutrición, servicio médico, cronometraje, tecnología. Se ubica en la categoría equivalente.',
    esPatrocinio: true,
    badgeClass: 'bg-stone-200 text-stone-700',
  },
  {
    slug: 'media_partner',
    label: 'Media Partner',
    subtitle: null,
    cupos: null,
    cuposLabel: 'libre',
    monto: null,
    montoLabel: 'Especie',
    incluye: 'Cobertura y difusión del evento: prensa, radio, televisión, podcast o medios digitales. Aporte en especie valorado a precio de mercado.',
    esPatrocinio: true,
    badgeClass: 'bg-sky-100 text-sky-800',
  },
  {
    slug: 'zona_marcas',
    label: 'Zona de Marcas',
    subtitle: null,
    cupos: null,
    cuposLabel: 'libre',
    monto: 3000,
    montoLabel: '3,000/ev',
    incluye: 'Stand para sampling. Flexible: RD$3,000/evento o RD$12,000 por los 5 eventos. No constituye patrocinio.',
    esPatrocinio: false,
    badgeClass: 'bg-neutral-100 text-neutral-600',
  },
];

export const NOTA_AL_PIE =
  'Precios de referencia sujetos a disponibilidad de cupos. ' +
  'Los aportes en especie se valoran a precio de mercado.';

export const getCategory = (slug) =>
  SPONSOR_CATEGORIES.find((c) => c.slug === slug) || null;

export const categoryLabel = (slug) => getCategory(slug)?.label || '';
